### PR TITLE
[12.0] [MIG] web_tree_image

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Display images and icons in tree view
=====================================

This module defines a tree image widget, to be used with either binary fields or (function) fields of type character. Use ``widget='image'`` in your view definition. Optionally, set a ``width`` attribute. Default width is 30px.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@JayVora-SerpentCS @pedrobaeza Can you please review it?

**Comment by pedrobaeza** [comment]
Odoo core now supports images in trees. The only missing part now is reading images from URLs, but this module is not needed as is.

**Comment by nikul-serpentcs** [comment]
@pedrobaeza In base image size very large.
Like This.
odoo 1




that's why this module here useful, In module set image size with tooltip.

**Comment by pedrobaeza** [comment]
@nikul-serpentcs but that's something you can limit size putting options and attributes AFAIK

**Comment by nikul-serpentcs** [comment]
> @nikul-serpentcs but that's something you can limit size putting options and attributes AFAIK

Yes, But Regarding v11 https://github.com/OCA/web/pull/1006#issuecomment-409858591 and https://github.com/OCA/web/pull/1006#issuecomment-409897414

we used size with tooltip 

CC @tarteo

**Comment by pedrobaeza** [comment]
@nikul-serpentcs but then we should light this module only for providing extra features, but not full one.

**Comment by tarteo** [comment]
I agree with @pedrobaeza the module only adds extra options and functionalities to the already existing image widget. This should be clear in the readme. 
The tooltip is nice but I don't like that the images are resized to a hard coded 30px this should be configurable.
Maybe rename the module to web_tree_image_option or web_tree_image_tooltip (as it only adds a tooltip).

**Comment by nikul-serpentcs** [comment]
Ok @tarteo 
Wait for @pedrobaeza or Other confirmation after we will be deciding what next

**Comment by pedrobaeza** [comment]
I agree with @tarteo

**Comment by nikul-serpentcs** [comment]
@pedrobaeza @tarteo Only rename the module to `web_tree_image_tooltip`?

Or add any Features

CC @JayVora-SerpentCS

**Comment by tarteo** [comment]
@nikul-serpentcs web_tree_image_tooltip and change the readme so it's clear it's an extension of the existing widget.

**Comment by nikul-serpentcs** [comment]
@tarteo @pedrobaeza  Renamed `web_tree_image_tooltip` module, Could you please review it.

File: web_tree_image_tooltip/i18n/web_tree_image.pot
Comment (sudhir-erpharbor) [inline web_tree_image_tooltip/i18n/web_tree_image.pot:6]: Odoo Server 12.0

Comment (nikul-serpentcs) [reply web_tree_image_tooltip/i18n/web_tree_image.pot:6]: @sudhir-erpharbor I think here @OCA-git-bot change automatically, So no need.

Comment (JayVora-SerpentCS) [reply web_tree_image_tooltip/i18n/web_tree_image.pot:6]: Yes weblate would do it.

Comment (sudhir-erpharbor) [reply web_tree_image_tooltip/i18n/web_tree_image.pot:6]: Sounds good.

Comment (pedrobaeza) [reply web_tree_image_tooltip/i18n/web_tree_image.pot:6]: Well, for being precise, it will be changed by Weblate when the first change in the translations is done, and it will be changed by a Weblate text, not Odoo version. Take into account that this text is ignored by default and it's only put as reference.

**Comment by mhdamjadenaya** [comment]
in progress.

**Comment by nikul-serpentcs** [comment]
https://github.com/OCA/web/pull/1100#issuecomment-438258619 Reminder :smile: 
@pedrobaeza @tarteo @sudhir-erpharbor

File: web_tree_image_tooltip/__manifest__.py
Comment (tarteo) [inline web_tree_image_tooltip/__manifest__.py]: This is not needed anymore.

File: web_tree_image_tooltip/README.rst
Comment (tarteo) [inline web_tree_image_tooltip/README.rst]: This is false. I don't see the handling of `width` in the Javascript. 

Use the new split readme: https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

File: web_tree_image_tooltip/__manifest__.py
Comment (tarteo) [inline web_tree_image_tooltip/__manifest__.py]: Use lower case characters for the repository url so it's consistent with all other modules. https://github.com/OCA/Web -> https://github.com/OCA/web

**Review by tarteo — CHANGES_REQUESTED** [review]
Can you fix the linter issues.

File: web_tree_image_tooltip/README.rst
Comment (nikul-serpentcs) [reply web_tree_image_tooltip/README.rst]: > Use the new split readme: https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

@tarteo It's required?

File: web_tree_image_tooltip/static/src/js/tooltip.js
Comment (alexey-pelykh) [inline web_tree_image_tooltip/static/src/js/tooltip.js]: It would be great if you'd normalized tabs vs spaces usage

File: web_tree_image_tooltip/view/assets.xml
Comment (alexey-pelykh) [inline web_tree_image_tooltip/view/assets.xml:2]: License/Copyright is option, yet appreciated

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]
Please squash/rearrange your last 4 commits according to the guidelines

File: web_tree_image_tooltip/README.rst
Comment (tarteo) [reply web_tree_image_tooltip/README.rst]: Don't know, but it looks much better.

Comment (nikul-serpentcs) [reply web_tree_image_tooltip/README.rst]: > Don't know, but it looks much better.

@tarteo Done, Yes it looks Good.

**Comment by nikul-serpentcs** [comment]
@tarteo @alexey-pelykh Done

File: web_tree_image_tooltip/view/assets.xml
Comment (nikul-serpentcs) [reply web_tree_image_tooltip/view/assets.xml:2]: > License/Copyright is option, yet appreciated

It's Not required

File: web_tree_image_tooltip/static/src/js/tooltip.js
Comment (tarteo) [inline web_tree_image_tooltip/static/src/js/tooltip.js]: Can you remove the back slash?

File: web_tree_image_tooltip/readme/DESCRIPTION.rst
Comment (tarteo) [inline web_tree_image_tooltip/readme/DESCRIPTION.rst]: I don't understand what this means. Do you mean 'Additionally, the default width is 30px'?

**Review by tarteo — CHANGES_REQUESTED** [review]

File: web_tree_image_tooltip/readme/DESCRIPTION.rst
Comment (nikul-serpentcs) [reply web_tree_image_tooltip/readme/DESCRIPTION.rst]: > Do you mean 'Additionally, the default width is 30px'?

@tarteo Yes, We Added a Image size.

File: web_tree_image_tooltip/static/src/js/tooltip.js
Comment (nikul-serpentcs) [reply web_tree_image_tooltip/static/src/js/tooltip.js]: @tarteo Done!

**Review by alexey-pelykh — APPROVED** [review]

File: web_tree_image_tooltip/readme/DESCRIPTION.rst
Comment (tarteo) [reply web_tree_image_tooltip/readme/DESCRIPTION.rst]: Can you try to clear up the description, please. I know what the module does, but I think for somebody that sees it in the app store it's not very clear.

Comment (nikul-serpentcs) [reply web_tree_image_tooltip/readme/DESCRIPTION.rst]: @tarteo Done, Added a description like this `This module defines images and icons in tree view via Tooltip. Additionally, Default width with 30px.`
If any doubt, then add your comment/description, So i will re-write that.

File: web_tree_image_tooltip/view/assets.xml
Comment (tarteo) [inline web_tree_image_tooltip/view/assets.xml]: Can you change the file location according to the new module directory name.

**Review by tarteo — CHANGES_REQUESTED** [review]
The description is still bad IMO. There still seems to be 4 linter errors, please fix those.

File: web_tree_image_tooltip/view/assets.xml
Comment (nikul-serpentcs) [reply web_tree_image_tooltip/view/assets.xml]: @tarteo Done

**Comment by nikul-serpentcs** [comment]
@pedrobaeza @tarteo @oca-clabot  All comment changes are Done, Can you please move forward.

**Comment by nikul-serpentcs** [comment]
> @pedrobaeza @tarteo @oca-clabot All comment changes are Done, Can you please move forward.

Ping

**Comment by TimLai125** [comment]
Hi @pedrobaeza @tarteo @nikul-serpentcs May I know if there is anything pending with this PR?

**Review by tarteo — APPROVED** [review]
@TimLai125 The README is still bad but the functionality is OK.

**Review by TimLai125 — COMMENTED** [review]
Might be a picky comment, the tooltip image is not showed for the first hover, e.g. https://drive.google.com/file/d/1A3Y-Ag4Z508uXTKmocBO9Cz6-yCWOroo/view

**Comment by tarteo** [comment]
> Might be a picky comment, the tooltip image is not showed for the first hover, e.g. https://drive.google.com/file/d/1A3Y-Ag4Z508uXTKmocBO9Cz6-yCWOroo/view

This seems the case, this needs to be fixed before merge unless this was also the case in 11.0.

File: web_tree_image_tooltip/static/src/js/tooltip.js
Comment (Tardo) [inline web_tree_image_tooltip/static/src/js/tooltip.js]: Here can force 'show' to fix onmouseover issue
```suggestion
            }).tooltip('show');
```

**Review by Tardo — CHANGES_REQUESTED** [review]

**Review by Tardo — APPROVED** [review]
LGTM :+1:

**Comment by legalsylvain** [comment]
Hi @nikul-serpentcs 

just a message to thank you for this excellent module, which covers exactly a recent need in my company.

**Pull request closed without merging**
